src/jit/lsra.cpp

```diff
@@ -8613,6 +8622,15 @@ void LinearScan::updateMaxSpill(RefPosition* refPosition)
                     ReturnTypeDesc* retTypeDesc = treeNode->AsCall()->GetReturnTypeDesc();
                     typ                         = retTypeDesc->GetReturnRegType(refPosition->getMultiRegIdx());
                 }
+#ifdef _TARGET_ARM_
+                else if (treeNode->OperIsPutArgReg())
+                {
+                    // For double arg regs, the type is changed to long since they must be passed via `r0-r3`.
+                    // However when they get spilled, they should be treated as separated int registers.
```

Comment:
I think that this is just the case for `ARM_SOFTFP`, right? In other cases, an actual `TYP_LONG` would have been decomposed, so it would be better to put this under `#ifdef ARM_SOFTFP`.